android/src/ti/bluetooth/central/TiBluetoothCentralManagerProxy.java

```diff
@@ -0,0 +1,232 @@
+import ti.bluetooth.peripheral.TiBluetoothPeripheralProxy;
+
+@Kroll.proxy(parentModule = TiBluetoothModule.class)
+public class TiBluetoothCentralManagerProxy
+    extends KrollProxy implements OnBluetoothStateChangedListener,
+                                  OnPeripheralConnectionStateChangedListener {
+  private static final String DID_UPDATE_STATE_EVENT = "didUpdateState";
+  private static final String DID_DISCOVER_PERIPHERAL = "didDiscoverPeripheral";
+  private static final String ERROR_SCAN_FAILED = "errorScanFailed";
+  private static final String DID_CONNECT_PERIPHERAL = "didConnectPeripheral";
+  private static final String DID_DISCONNECT_PERIPHERAL =
+      "didDisconnectPeripheral";
+  private static final String DID_FAIL_TO_CONNECT_PERIPHERAL =
+      "didFailToConnectPeripheral";
+  private static final String NOTIFY_ON_CONNECTION_KEY = "notifyOnConnection";
+  private static final String NOTIFY_ON_DISCONNECTION_KEY =
+      "notifyOnDisconnection";
+
+  private static final int SCAN_MODE_BALANCED =
+      TiBluetoothModule.SCAN_MODE_BALANCED;
+  private static final int SCAN_MODE_LOW_LATENCY =
+      TiBluetoothModule.SCAN_MODE_LOW_LATENCY;
+  private static final int SCAN_MODE_LOW_POWER =
+      TiBluetoothModule.SCAN_MODE_LOW_POWER;
+  private static final int SCAN_MODE_OPPORTUNISTIC =
+      TiBluetoothModule.SCAN_MODE_OPPORTUNISTIC;
+
+  private Context context;
+  private BluetoothAdapter bluetoothAdapter;
+  private int scanMode;
+  private int bluetoothState;
+
+  private boolean isScanning;
+
+  public TiBluetoothCentralManagerProxy(Context context,
+                                        BluetoothAdapter bluetoothAdapter) {
+    this.context = context;
+    this.bluetoothAdapter = bluetoothAdapter;
+    this.bluetoothState = bluetoothAdapter.getState();
+    this.scanMode = ScanSettings.SCAN_MODE_BALANCED;
+
+    IntentFilter intentFilter = new IntentFilter();
+    intentFilter.addAction(BluetoothAdapter.ACTION_STATE_CHANGED);
+
+    context.registerReceiver(new TiBluetoohBroadcastReceiver(this),
+                             intentFilter);
+  }
+
+  @Kroll.method
+  public boolean isScanning() {
+    return isScanning;
+  }
+
+  @Kroll.method
+  public void startScan() {
+    startScanWithServices(null);
+  }
+
+  @Kroll.method
+  public void startScanWithServices(String[] services) {
+    ScanSettings settings =
+        new ScanSettings.Builder().setScanMode(scanMode).build();
+
+    List<ScanFilter> filters = buildScanFilters(services);
+    bluetoothAdapter.getBluetoothLeScanner().startScan(filters, settings,
+                                                       scanCallback);
+    isScanning = true;
+  }
+
+  @Kroll.method
+  public void stopScan() {
+    bluetoothAdapter.getBluetoothLeScanner().stopScan(scanCallback);
+    isScanning = false;
+  }
+
+  @Kroll.method
+  public void connectPeripheral(TiBluetoothPeripheralProxy peripheral,
+                                KrollDict properties) {
```

Comment:
the properties should be `(optional = true)`

Reply:
Done!